### ISSUE TITLE
[FEATURE] Writing list in single cell from data list

### ISSUE THREAD

**Issue by EsmeeIDEMS** [IDEMSInternational/parenting-app-ui#1467]
**What?**
I would like to be able to write a cell entry like 

```
name: item_1 | text: first; name: item_2 | text: second; name: item_3 | text: third
```
or 
```
 first; second; third
```
from a data list of the form

| id     | text                |
| --------- | ------                      |
|item_1| first                |
|item_2| second	        |
|item_3| third       |

**Why?**
Useful for creating an answer list (for a radio group or combo box) based on what's in a data list. 

**How?**
Possibly using an items environment. This does not work currently: 
1. See [example_items_radio](https://docs.google.com/spreadsheets/d/1n82prWaYx-HWKW0ibxv-sO_Vuw9WxvuNLRJ_jI_WXtk/edit#gid=1427752706).
<img width="273" alt="image" src="https://user-images.githubusercontent.com/74557272/182905997-f305f721-dfaa-42ce-8b70-0aa8c58d6a93.png"><img width="395" alt="image" src="https://user-images.githubusercontent.com/74557272/182906055-fdc8f281-25cf-49f7-9757-5711a11cb00d.png">
2. See [example_items_nav](https://docs.google.com/spreadsheets/d/1n82prWaYx-HWKW0ibxv-sO_Vuw9WxvuNLRJ_jI_WXtk/edit#gid=1360411654) 
<img width="307" alt="image" src="https://user-images.githubusercontent.com/74557272/182907904-6c4c2c96-aea6-4bfc-b3ca-26a8764b8ed5.png"><img width="466" alt="image" src="https://user-images.githubusercontent.com/74557272/182907954-acaa552e-4568-4438-88d7-c4683386a818.png">

**Comment by EsmeeIDEMS** [comment]
On further thought I decided I could also go around this issue by authoring an extra cell in the data list which uses an excel formula to concatenate the strings (separated by `;`s). 

However, that does not work - see example added to [example_items_nav](https://docs.google.com/spreadsheets/d/1n82prWaYx-HWKW0ibxv-sO_Vuw9WxvuNLRJ_jI_WXtk/edit#gid=1360411654)
<img width="212" alt="image" src="https://user-images.githubusercontent.com/74557272/184341720-d1b3b66b-cde1-42be-b38b-7bb31a6a25f8.png">

**Comment by chrismclarke** [comment]
@EsmeeIDEMS 
I took a quick look but think authoring as part of an item loop may be a bit confusing at the best of time and (hopefully) not really necessary once we have better ways to natively filter/append data lists so I think I might suggest applying a `wont fix` label to this for now

That being said, it would be good to have a solution that works for what you need right now. I'm not sure if you already tried this and there might be something I'm missing, but simply adding another column to the data list so that `name` and `text` both exist allow it to be called directly in the template, i.e.
```
answer_list: @data.example_items_radio
```
(this was likely not previously working but fixed in #1460)
I've updated the examples to show it working with the simple syntax

![image](https://user-images.githubusercontent.com/10515065/184620057-3caff9d5-2b1b-46af-88bc-a4f3cc11735e.png)

**Comment by jfmcquade** [comment]
As i understand it, Chris's workaround is good for components that use `answer_list`s, but another workaround would still be useful for the case of `nav_group`s.

I've tried to diagnose the issues with the proposed workarounds and then have suggested an alternative route:
- `nav_group` needs to receive an array of template names in it `value`
- The item loop method above doesn't work because it creates nested arrays
  - e.g. [[["example_text_button"], "example_text_image"], "example_text"]
  - We could flatten this array within `nav_group` code, but doesn't seem like the best workaround
- Using the concat method above doesn't work because it creates an array containing a single string, rather than just a string which would be correctly parsed as an array.
  - e.g. ['example_text_button; example_text_image; example_text'] – since the string is inside an array it is missed by the template parser so remains a string and not an array

- If we're coding a specific workaround, why not make this as simple as possible from the authoring level, and allow `nav_group` to be passed a whole `data_list` with a `template` column, and parse the template names from that object.
  - That's what I've done in PR #1487
